src/nfa/nfa-intersection.cc

```diff
@@ -368,7 +368,7 @@ void intersection(Nfa* res, const Nfa &lhs, const Nfa &rhs, Symbol epsilon, Prod
     *res = intersection.get_product();
 }
 
-Nfa intersection(const Nfa& lhs, const Nfa& rhs, const Symbol epsilon, ProductMap*  prod_map)
+Nfa intersection_over_epsilon(const Nfa& lhs, const Nfa& rhs, Symbol epsilon, ProductMap*  prod_map)
```

Comment:
Why do you remove the const qualifiers here? Once the epsilon is specified, it will remain const for the whole duration of the function execution.

Reply:
By mistake